### ISSUE TITLE
Add support for custom types to "typeorm-typescript/enforce-column-types"

### ISSUE THREAD

**Issue by SimonSimCity** [daniel7grant/eslint-plugin-typeorm-typescript#5]
I'd love to get an option to define things like `import { UUID } from 'crypto';` as compatible to `string`. This way I could take advantage of the null-checking here!

**Comment by daniel7grant** [comment]
I detailed in another comment (https://github.com/daniel7grant/eslint-plugin-typeorm-typescript/issues/4#issuecomment-2128041022) what is the issue with aliases. However I could imagine an option in the ESLint configuration, where you could add custom aliases:

```json
{
  "rules": {
    "typeorm-typescript/enforce-column-types": ["error", {
      "customTypes": {
        "UUID": "string"
      }
    }]
  }
}
```

It would be riskier, but there could even be a built-in mapping of well-known types, that is validated automatically (for example somebody redefining UUID to store it as an object). However I'd prefer to get this first as an opt-in feature.

Would this be what you had in mind?

**Comment by SimonSimCity** [comment]
This idea here could be an intermediate step towards full type support (as I suggested in my comment on #4). It's not quite type-safe, but I guess that's something I'd have to live with then 😅 My database-layer anyways won't be 100% type safe, because I cannot create a rule by which it would check for my uuid pattern, so I'll always have to rely on e.g. that my application is the only thing modifying the database.

**Comment by daniel7grant** [comment]
I added some support for type safety in [v0.3.0](https://github.com/daniel7grant/eslint-plugin-typeorm-typescript/releases/tag/v0.3.0) so it should now handle aliases (your UUID example should work as well). Thank you for your suggestion!

**Comment by norbornen** [comment]
> ```json
>     "typeorm-typescript/enforce-column-types": ["error", {
>       "customTypes": {
>         "UUID": "string"
>       }
>     }]
> ```

What about this?
My database driver (mysql) cast `bigint` and `decimal` to `string`...

**Comment by norbornen** [comment]
This may depend on both the internal default settings of typeorm and the specific settings [of the driver](https://github.com/mysqljs/mysql/blob/dc9c152a87ec51a1f647447268917243d2eab1fd/Readme.md?plain=1#L1402) set by the developer.

Example for typeorm usage with custom `bigint` handler:
```
const pg = require('pg');

pg.types.setTypeParser(20, parseInt); // для BigInt

/** @type {import('typeorm/driver/postgres/PostgresConnectionOptions').PostgresConnectionOptions} */
const ormconfig = {
  type: "postgres",
  ...
};

module.exports = ormconfig;
```

**Comment by daniel7grant** [comment]
I tested Typeorm with the most popular drivers (pg, mysql2, sqlite3), and I found that you are indeed correct. It gives different outputs for different drivers:

|         | pg     | mysql2 | sqlite3 |
|---------|--------|--------|---------|
| `bigint`  | string | string | number  |
| `decimal` | string | string | number  |

I would guess this is based on SQLite's [lack of different sized int and float columns](https://sqlite.org/datatype3.html).

I'm not sure how to proceed, I'm wary of adding new configurations for every change, which would nudge me towards accepting both string and number. On the other hand, that would add to the confusion that this library is supposed to solve. Do you think a `driver` configuration with `sqlite` option and moving the default value to string would solve this problem?

**Comment by daniel7grant** [comment]
Hey @norbornen, I created a new PR #19, which changes the default behaviour to follow your use-case. Can you follow up, if this is fixed for you? I can merge it for the next release.

**Comment by daniel7grant** [comment]
I released the fix for bigint and decimal in version [v0.5.0](https://github.com/daniel7grant/eslint-plugin-typeorm-typescript/releases/tag/v0.5.0). If you have any more issues, feel free to reopen.

Thanks for your help!
